pkg/ebpf/state.go

```diff
@@ -184,11 +204,19 @@ func (ns *networkState) StoreClosedConnection(conn ConnectionStats) {
 	for _, client := range ns.clients {
 		// If we've seen this closed connection already, lets combine the two
 		if prev, ok := client.closedConnections[string(key)]; ok {
-			// We received either the connections either out of order, or it's the same one we've already seen.
-			// Lets skip it for now.
+			// Check if we did not receive the same close event twice, if it's the case we only want to keep one
+			if prev.id == conn.id {
+				// Let's keep the one with the highest stats since we can't rely on the last update epoch
+				// See: https://github.com/weaveworks/scope/issues/2650
+				client.closedConnections[string(key)] = connWithHigherStats(prev, conn)
+				ns.telemetry.duplicateCloseEvents++
+				continue
+			}
+
+			// We received the two connections out of order.
+			// Increment the telemetry for unordered connections
 			if prev.LastUpdateEpoch >= conn.LastUpdateEpoch {
 				ns.telemetry.unorderedConns++
```

Comment:
why did the `continue` go away? We should be seeing this less with this change, but if this condition is true, the current closed connection should still be ignored.

Reply:
I don't think it should, since their id is different we know that those 2 close events represent 2 different connections hence they should both be counted (and aggregated)